cmd/vic-machine/create/create.go

```diff
@@ -864,17 +875,47 @@ func (c *Create) processDNSServers() error {
 	return nil
 }
 
-func (c *Create) processVolumeStores() error {
+func (c *Create) processDefaultVolumeStore() error {
 	defer trace.End(trace.Begin(""))
+	//we make this map here since it is used many times after.
 	c.VolumeLocations = make(map[string]string)
+	if c.DefaultVolumeLocation != "" {
+		c.VolumeLocations[defaultVolumeLabel] = c.DefaultVolumeLocation
+	}
+	return nil
+}
+
+func (c *Create) processVolumeStores() error {
+	defer trace.End(trace.Begin(""))
 	for _, arg := range c.volumeStores {
+
 		splitMeta := strings.SplitN(arg, ":", 2)
 		if len(splitMeta) != 2 {
 			return errors.New("Volume store input must be in format datastore/path:label")
 		}
-		c.VolumeLocations[splitMeta[1]] = splitMeta[0]
+		label := splitMeta[1]
+		path := splitMeta[0]
+
+		switch label {
+		case defaultVolumeLabel:
+			otherPath, ok := c.VolumeLocations[label]
+			if !ok {
+				c.VolumeLocations[label] = path
+				continue
+			}
+			if path != otherPath {
+				return errors.Errorf("Multiple paths were tagged for the same label(%s), volumestore labels can only have one distinct path. The preferred method of setting the default store is the --default-volume-store flag.", label)
+			}
+		default:
+			if _, ok := c.VolumeLocations[label]; ok {
+				return errors.Errorf("Multiple paths were tagged for the same label(%s), volumestore labels can only have one distinct path.", label)
+			}
+			c.VolumeLocations[label] = path
```

Comment:
Yeah I actually wanted to make a function that I just passed a string too. I think I will do that.